source/includes/data-modeling/persistence-configuration.rb

```diff
@@ -0,0 +1,130 @@
+# start default modeling
+class Restaurant
+    include Mongoid::Document
+end
+
+class Person
+    include Mongoid::Document
+end
+# end default modeling
+
+# start set pluralization
+ActiveSupport::Inflector.inflections do |inflect|
+    inflect.plural("rey", "reyes")
+end
+# end set pluralization
+
+# start BSON model
+{
+    "_id" : ObjectId("4d3ed089fb60ab534684b7e9"),
+    "title" : "Sir",
+    "name" : {
+        "_id" : ObjectId("4d3ed089fb60ab534684b7ff"),
+        "first_name" : "Durran"
+    },
+    "addresses" : [
+        {
+          "_id" : ObjectId("4d3ed089fb60ab534684b7e0"),
+          "city" : "Berlin",
+          "country" : "Deutschland"
+        }
+    ]
+}
+# end BSON model
+
+# start store_in example
+class Person
+    include Mongoid::Document
+    store_in collection: "citizens", database: "other", client: "analytics"
+end
+# end store_in example
+
+# start store_in lambda example
+class Band
+    include Mongoid::Document
+    store_in database: ->{ Thread.current[:database] }
+end
+# end store_in lambda example
+
+# start persistence context attributes
+puts Band.client_name
+
+puts Band.database_name
+
+puts Band.collection_name
+# end persistence context attributes
+
+# start with() example
+class Band
+    include Mongoid::Document
+
+    field :name, type: String
+    field :likes, type: Integer
+end
+
+# Creates document in 'bands' collection in 'music-non-stop' database within
+# default cluster
+Band.with(database: "music-non-stop") do |band_class|
+    band_class.create(name: "Medusa and the Polyps")
+end
+
+# Deletes all documents in 'artists' collection within default database
+Band.with(collection: "artists") do |band_class|
+    band_class.delete_all
+end
+
+band = Band.new(name: "Japanese Breakfast")
+
+# Perform operations on tertiary cluster
+band.with(client: :tertiary) do |band_object|
+    band_object.save!
+    
+    band.save!
+end
+# end with() example
+
+# start read configuration
+Band.with(read: {mode: :secondary}) do
+    Band.count
+
+    # Write operation runs in default 
+    # persistence context
```

Comment:
As mentioned previously, this write operation _also runs in the new context_. But because the new context only differs from the old context by the specification of a read preference, the write operation will behave the same on both the new context and the default context, but _technically_, the write is running with the new context.

```suggestion
    # Write operation runs in the new persistence context,
    # but will not be affected by the read preference.
```